packages/epo-widget-lib/src/widgets/CameraFilter/SpectrumDisplay/SpectrumLabels/SpectrumLabels.tsx

```diff
@@ -0,0 +1,74 @@
+import { FunctionComponent } from "react";
+import { useTranslation } from "react-i18next";
+import { Band } from "@/types/astro";
+import * as Styled from "../../styles";
+import ColorLabels from "../ColorLabels";
+import { Spectrum } from "../../data";
+import { intersection } from "@/lib/utils";
+
+interface SpectrumLabelsProps {
+  spectrums: Spectrum[];
+  activeBand?: Band;
+  min: number;
+  max: number;
+  activeMin?: number;
+  activeMax?: number;
+  isCondensed: boolean;
+}
+
+const SpectrumLabels: FunctionComponent<SpectrumLabelsProps> = ({
+  spectrums,
+  activeBand,
+  min,
+  max,
+  activeMin = 0,
+  activeMax = 0,
+  isCondensed,
+}) => {
+  const { t } = useTranslation();
+  const spectrumLabel = t("camera_filter.labels.captured_range", {
+    context: activeBand ? "" : "no_filter",
+    filter: activeBand,
+  });
+  return (
+    <g
+      role="list"
+      aria-label={spectrumLabel}
+      aria-live="polite"
+      data-testid="spectrum-labels"
+    >
+      {spectrums.map(({ name, upper, lower, colors }) => {
+        const safeUpper = upper || max;
+        const safeLower = lower || min;
+        const isHidden =
+          activeBand &&
+          intersection([safeLower, safeUpper], [activeMin, activeMax]) === null;
+        return (
+          <g role="listitem" aria-hidden={isHidden} key={name}>
```

Comment:
these are the labels "ultraviolet light" "visible light" "infrared light", they are added and removed from the accessibility tree based on the shown range. If visible light is totally hidden, all color labels hide too.